server/src/main/java/org/elasticsearch/cluster/coordination/ElectionSchedulerFactory.java

```diff
@@ -172,6 +172,12 @@ void scheduleNextElection(final TimeValue gracePeriod, final Runnable scheduledR
             final long maxDelayMillis = Math.min(maxTimeout.millis(), initialTimeout.millis() + thisAttempt * backoffTime.millis());
             final long delayMillis = toPositiveLongAtMost(random.nextLong(), maxDelayMillis) + gracePeriod.millis();
             final Runnable runnable = new AbstractRunnable() {
+
+                @Override
+                public void onRejection(Exception e) {
+                    logger.debug("threadpool was shut down", e);
```

Comment:
Yea the problem was that inside `ThreadedRunnable.run` we just call:

```java
                executor.execute(runnable);
```

If this is called with an `AbstractRunnable` then we will call `onRejection` on it instead of throwing the rejection and the handling in `ThreadedRunnable` doesn't come into play => we need to manually catch this case here.